src/main/java/org/hobbit/core/containerservice/ContainerCreation.java

```diff
@@ -0,0 +1,21 @@
+package org.hobbit.core.containerservice;
+
+import org.hobbit.core.components.AbstractCommandReceivingComponent;
+/**
+ * ContainerCreation provides the facility to implement the functionalities 
+ * to create {@link DirectContainerCreator} or {@link RabbitMQContainerCreator}
+ * @author altaf, sourabh, yamini, melisa
+ *
+ */
+public interface ContainerCreation {
+	
+	void createDataGenerators(String dataGeneratorImageName, int numberOfDataGenerators,
+            String[] envVariables, AbstractCommandReceivingComponent dummyComponent);
+	
+	void createTaskGenerators(String taskGeneratorImageName, int numberOfTaskGenerators,
+            String[] envVariables, AbstractCommandReceivingComponent dummyComponent);
+	
+	void createEvaluationStorage(String evalStorageImageName, String[] envVariables,
+			AbstractCommandReceivingComponent dummyComponent);
```

Comment:
Why these three methods? It does not make much sense to me because of two different reasons:
1. For the creation of containers, it doesn't matter which container is created. To this end, the factory should only provide the functionality to create any container regardless which container should be created.
2. With this implementation, you are not able to create any other container. This limits the usage of the class a lot.

Apart from that, the container IDs should be returned by the method so that the requesting class can decide where it may want to store the ID.